homeassistant/components/snapcast/config_flow.py

```diff
@@ -0,0 +1,71 @@
+"""Snapcast config flow."""
+
+from __future__ import annotations
+
+import logging
+import socket
+
+import snapcast.control
+from snapcast.control.server import CONTROL_PORT
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SNAPCAST_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_PORT, default=CONTROL_PORT): int,
+    }
+)
+
+
+class SnapcastConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Snapcast config flow."""
+
+    async def async_step_user(self, user_input=None) -> FlowResult:
+        """Handle first step."""
+
+        def _show_form(errors=None):
+            return self.async_show_form(
+                step_id="user",
+                data_schema=SNAPCAST_SCHEMA,
+                errors=errors,
+            )
+
+        if not user_input:
+            return _show_form()
```

Comment:
This feels really disjointed and makes it hard to read.

Please use a simpler logic:
```python
    errors = {}
    if user_input:
        try:
            ...
        except:
            ...
        else:
            return self.async_create_entry(...)

    return self.async_show_form(...)
```

Reply:
Thanks. Is much better now.